### PR TITLE
os-tenant-networks

### PR DESCRIPTION
**Pull request by jtopjian**
This commit adds support for the os-tenant-networks extension.

### CODE REVIEW

**Comment by jtopjian** [comment]
While investigating #381, I came across the [os-tenant-networks](https://github.com/openstack/nova/blob/master/nova/api/openstack/compute/contrib/os_tenant_networks.py) extension. This looks to be a better solution to the problem I was trying to solve with #381 which is the ability for a tenant to retrieve basic information about the networks they have access to. 

This extension works with both nova-network and Neutron, though it's more beneficial to nova-network users. Neutron does not return the CIDR field, but could still be used as an alternative/easier way of mapping networks to UUIDs. 

This commit implements the List and Get actions, but not the Create action. I could implement the Create action, but I'll need a special environment to do this. Just like with #381, I don't anticipate Create being used often (or at all) with this extension.

**Comment by jtopjian** [comment]
Also, I'm not sure why the build is failing. Running the tests locally are passing. The logs seem to point to something with the networking/v2 API? Maybe I'm reading it wrong.

File: openstack/compute/v2/extensions/tenantnetworks/doc.go [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Package tenantnetwork provides the ability for tenants to see information about the networks they have access to
```
Comment (jrperritt) [inline openstack/compute/v2/extensions/tenantnetworks/doc.go]: /tenantnetwork/tenantnetworks

File: openstack/compute/v2/extensions/tenantnetworks/results.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+// A Network represents a nova-network that an instance communicates on
+type Network struct {
+	// CIDR is the IPv4 subnet.
+	CIDR string `mapstructure:"cidr"`
+
+	// ID is the UUID of the network.
+	ID string `mapstructure:"id"`
+
+	// Label is the common name that the network has..
```
Comment (jrperritt) [inline openstack/compute/v2/extensions/tenantnetworks/results.go]: One too many periods.

File: openstack/compute/v2/extensions/tenantnetworks/results.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	Label string `mapstructure:"label"`
```
Comment (jrperritt) [inline openstack/compute/v2/extensions/tenantnetworks/results.go]: Can we call this `Name` for consistency?

**Comment by jrperritt** [comment]
Looks good. Some very minor changes from me. I'm not worried about the build failure in Travis; it looks unrelated to this.

**Comment by jtopjian** [comment]
Cool - thanks for the review. Please see latest version.

**Comment by jrperritt** [comment]
Just one more

**Comment by jrperritt** [comment]
:+1:

**Pull request merged**